pkg/networkservice/common/policyroute/server.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2022 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policyroute
+
+import (
+	"context"
+
+	"sync"
+	"sync/atomic"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"gopkg.in/yaml.v2"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/fs"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+type policyrouteServer struct {
+	ctx        context.Context
+	policies   atomic.Value
+	once       sync.Once
+	configPath string
+}
+
+func (p *policyrouteServer) init(ctx context.Context) {
+	p.policies.Store([]*networkservice.PolicyRoute{})
+	updatePrefixes := func(bytes []byte) {
+		if bytes == nil {
+			p.policies.Store([]*networkservice.PolicyRoute{})
+		}
+		var source []*networkservice.PolicyRoute
+		err := yaml.Unmarshal(bytes, &source)
+		if err != nil {
+			log.FromContext(ctx).Errorf("Cannot unmarshal policies, err: %v", err.Error())
+			return
+		}
+		p.policies.Store(source)
+	}
+	updateCh := fs.WatchFile(p.ctx, p.configPath)
+	updatePrefixes(<-updateCh)
+	go func() {
+		for update := range updateCh {
+			updatePrefixes(update)
+		}
+	}()
+}
+
+// NewServer creates a NetworkServiceServer that will put the routing policies to connection context
+func NewServer(ctx context.Context, setters ...ServerOption) networkservice.NetworkServiceServer {
+	server := &policyrouteServer{
+		configPath: "/etc/route-policies/config.yaml",
```

Comment:
why are we reading these from a file in the chain element rather than passing them in as an object as an argument to NewServer?

Reply:
Because we assume that we can use `ConfigMap` and it can be changed. In this case we `watch` the config file to receive updates in a timely manner.

Reply:
@glazychev-art ConfigMaps should be read in main.go, not in a chain element.  I'm fine with main.go reading a config file... but a chain element shouldn't.

Reply:
@edwarnicke Configmap can be changed dynamically by the user. So the question, will we ignore the user's config map update?

Reply:
@denis-tingaikin If we want to support dynamic updating (which I'm fine with) we should still handle it via an object passed in, embedding config file watching logic inside a chain element ties that chain element to choices around the file system being the source of the config.  Best to avoid that.

Reply:
Agreed. Can we then pass _channel_ with policies into the chain element?

Reply:
@denis-tingaikin That's certainly one option... but we need to be careful to correctly handle cases where we don't receive policyies etc... blocking isn't OK in those cases, but errors would be.

Reply:
@edwarnicke Hmm, I think it makes sense, working with a channel can produce a complex logic in the chain element.

So as an alternative option we can provide some interface for getting actual PBRs

```go
type Provider interface {
 Get() []*networkservice.PolicyRoute
}
```

Thoughts?

Reply:
Or even simpler:

```go
type PolicyRoutesFunc func() []*networkpolicy.PolicyRoute
```